flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/DatabaseCalciteSchema.java

```diff
@@ -143,7 +143,8 @@ private Table convertConnectorTable(
 			return new TableSourceTable<>(
 					tableSource,
 					isStreamingMode,
-					FlinkStatistic.builder().tableStats(tableStats).build());
+					FlinkStatistic.builder().tableStats(tableStats).build(),
+					null);
```

Comment:
Will we also have `CatalogTable` for such TableSource wrapped table in the future? If not, I would suggest to change `CatalogTable` to `Opion[CatalogTable]` in `TableSourceTable`.

Reply:
Actually, I have some concern to put the whole `CatalogTable` in `TableSourceTable`. Because in CatalogTable, it may contains all the computed columns defined in DDL. But not all of them are retained in source, some of them may be applied to the following `LogicalProject`. Could we only put the information we need in `TabeSourceTable` and `TableSinkTable`? e.g. `List<String> partitionKeys`.

Reply:
I would say let's use the information from CatalogTable more carefully then... I can imagine partition keys are just a starter, we might need more and more information from CatalogTable in the future. CatalogTable is some kind of meta information about the table.

Reply:
Hi @KurtYoung , After temporary table support, every table should have `CatalogTable`. I think we can keep not Option.
Hi @wuchong , I agree with kurt, Actually, source may also need to know compute columns. It needs to know which fields it does not read. (except compute columns expressions)

Reply:
> source may also need to know compute columns. It needs to know which fields it does not read. (except compute columns expressions)
Actually, we will retain some expression in source (for rowtime generation), then we have to have another field in `TableSourceTable` to keep this information. say `generatedExpressions`, this might be confused with the expressions in `CatalogTable`?

Reply:
Hi Jark, I don't quite understand the expression in this source, but I think we need a good name to clarify it. It's may not just the `TableSourceTable` that saves them together? I think they are all attributes of table.